gwctl/pkg/printer/backends.go

```diff
@@ -28,7 +34,96 @@ import (
 )
 
 type BackendsPrinter struct {
-	Out io.Writer
+	Out   io.Writer
+	Clock clock.Clock
+}
+
+func (bp *BackendsPrinter) Print(resourceModel *resourcediscovery.ResourceModel) {
+	tw := tabwriter.NewWriter(bp.Out, 0, 0, 2, ' ', 0)
+	row := []string{"NAMESPACE", "NAME", "TYPE", "REFERRED BY ROUTES", "AGE", "POLICIES"}
+	tw.Write([]byte(strings.Join(row, "\t") + "\n"))
+
+	backendNodes := make([]*resourcediscovery.BackendNode, 0, len(resourceModel.Backends))
+	for _, backendNode := range resourceModel.Backends {
+		backendNodes = append(backendNodes, backendNode)
+	}
+
+	sort.Slice(backendNodes, func(i, j int) bool {
+		if backendNodes[i].Backend.GetNamespace() != backendNodes[j].Backend.GetNamespace() {
+			return backendNodes[i].Backend.GetNamespace() < backendNodes[j].Backend.GetNamespace()
+		}
+		return backendNodes[i].Backend.GetName() < backendNodes[j].Backend.GetName()
+	})
+
+	httpRouteNodes := make([]*resourcediscovery.HTTPRouteNode, 0, len(resourceModel.HTTPRoutes))
+	for _, httpRouteNode := range resourceModel.HTTPRoutes {
+		httpRouteNodes = append(httpRouteNodes, httpRouteNode)
+	}
+
+	for _, backendNode := range backendNodes {
+		backend := backendNode.Backend
+
+		parentHttpRoutes := []string{}
+		remainderHttpRoutes := 0
+		associatedDistinctPolicies := map[string]any{}
+
+		// the nested loops could've been made more efficient (on textbook) by using map between httpRouteNodes and backendRefs
+		// and leveraging the O(1) lookups but that seemed to sacrifice the code readability/structure, so didn't choose to do so
+		for _, httpRouteNode := range httpRouteNodes {
+			httpRoute := httpRouteNode.HTTPRoute
+
+			for _, backendRefNode := range httpRouteNode.Backends {
+				backendRef := backendRefNode.Backend
+
+				if backendRef.GetName() != backend.GetName() ||
+					backendRef.GetNamespace() != backend.GetNamespace() ||
+					backendRef.GetKind() != backend.GetKind() {
+					continue
+				}
+
+				if len(parentHttpRoutes) < 2 {
+					namespacedName := client.ObjectKeyFromObject(httpRoute).String()
+					parentHttpRoutes = append(parentHttpRoutes, namespacedName)
+				} else {
+					remainderHttpRoutes += 1
+				}
+
+				for _, effectivePolicy := range httpRouteNode.EffectivePolicies {
```

Comment:
The space around inherited policies is somewhat still under changes. For the time being, let's simply imply the count of policies to be the policies that are directly attached to this Backend. This would mean simply counting the number of elements in the `BackendNode.Policies`